File: Java/src/Unique Paths.java
import java.util.Arrays;

public class Solution
{
  public int uniquePaths(int m, int n)
  {
    if (m == 0 || n == 0)
      return 0;
    if (m == 1 || n == 1)
      return 1;

    int[][] grid = new int[n][m];
    for (int[] row : grid)
    {
      Arrays.fill(row, 0);
    }

    Arrays.fill(grid[0], 1);
    for (int j = 0; j < n; j++)
      grid[j][0] = 1;

    for (int j = 1; j < n; j++)
    {
      for (int i = 1; i < m; i++)
      {
        grid[j][i] = grid[j - 1][i] + grid[j][i - 1];
      }
    }
    return grid[n - 1][m - 1];
  }
}

File: Python/Unique Paths.py

class Solution:
    # @return an integer
    def uniquePaths(self, m, n):
        if m == 0 or n == 0:
            return 0
        if m == 1 or n == 1:
            return 1
        grid = [[0] * m] * n 
        
        grid[0] = [1]* m
        
        for j in range(n):
            grid[j][0] = 1
        
        for i in range(1,n):
            for j in range(1,m):
                grid[i][j] = grid[i-1][j] + grid[i][j-1]
        
        return grid[n-1][m-1]